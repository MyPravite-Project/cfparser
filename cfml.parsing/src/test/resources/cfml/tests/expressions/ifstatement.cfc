if ( Now() gt dateadd("d",  -val(iDiff) , startDate) ){
}