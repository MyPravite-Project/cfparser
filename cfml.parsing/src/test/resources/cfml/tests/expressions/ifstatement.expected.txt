/*===TOKENS===*/
IF                       <if>
'('                      <(>
IDENTIFIER               <Now>
'('                      <(>
')'                      <)>
GT                       <gt>
IDENTIFIER               <dateadd>
'('                      <(>
OPEN_STRING              <">
STRING_LITERAL           <d>
CLOSE_STRING             <">
','                      <,>
'-'                      <->
IDENTIFIER               <val>
'('                      <(>
IDENTIFIER               <iDiff>
')'                      <)>
','                      <,>
IDENTIFIER               <startDate>
')'                      <)>
')'                      <)>
'{'                      <{>
Hidden:NEWLINE                  <>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (element
    (statement
      (ifStatement
        if
        (condition
          (
          (baseExpression
            (baseExpression
              (unaryExpression
                (memberExpression (functionCall (identifier Now) ( argumentList )))
              )
            )
            (compareExpressionOperator gt)
            (baseExpression
              (unaryExpression
                (memberExpression
                  (functionCall
                    (identifier dateadd)
                    (
                    (argumentList
                      (argument
                        (startExpression
                          (baseExpression
                            (unaryExpression
                              (primaryExpression
                                (literalExpression (stringLiteral " (stringLiteralPart d) "))
                              )
                            )
                          )
                        )
                      )
                      ,
                      (argument
                        (startExpression
                          (baseExpression
                            -
                            (baseExpression
                              (unaryExpression
                                (memberExpression
                                  (functionCall
                                    (identifier val)
                                    (
                                    (argumentList
                                      (argument
                                        (startExpression
                                          (baseExpression (unaryExpression (memberExpression (identifier iDiff))))
                                        )
                                      )
                                    )
                                    )
                                  )
                                )
                              )
                            )
                          )
                        )
                      )
                      ,
                      (argument
                        (startExpression
                          (baseExpression (unaryExpression (memberExpression (identifier startDate))))
                        )
                      )
                    )
                    )
                  )
                )
              )
            )
          )
          )
        )
        (statement (compoundStatement { }))
      )
    )
  )
)
/*======*/
/*===DECOMPILE===*/
if(Now() gt dateadd('d', -val(iDiff), startDate) )   {

  }
/*======*/